sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -148,10 +148,8 @@ abstract class Expression extends TreeNode[Expression] {
    * Checks the input data types, returns `TypeCheckResult.success` if it's valid,
    * or returns a `TypeCheckResult` with an error message if invalid.
    * Note: it's not valid to call this method until `childrenResolved == true`
-   * TODO: we should remove the default implementation and implement it for all
-   * expressions with proper error message.
    */
-  def checkInputDataTypes(): TypeCheckResult = TypeCheckResult.TypeCheckSuccess
+  def checkInputDataTypes(): TypeCheckResult
```

Comment:
Do we really want this? There are a lot of expressions that don't need type check...